src/Documentation/Frequently-Asked-Questions.md

```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
+That may be a sign of the grain being too large, and decomposing it into a set of smaller grains may lead to a more stable and balanced system.
+
+An application entity may be too small to be a grain if that would cause constant interaction of other grains with it, and as a result, to too much of messaging overhead.
+In such cases, it may make more sense to make those closely interacting entities part of a single grain, so that they would invoke each other directly.
+
+### How should you avoid grain hot spots?
 
 The throughput of a grain is limited by a single thread that its activation can execute on.
-Therefore, it is advisable to avoid designs where a single grain receives a disproportionate share of requests.
+Therefore, it is advisable to avoid designs where a single grain receives a disproportionate share of requests or is involved in processing requests to other grains.
 There are various patterns that help prevent overloading of a single grain even when logically it is a central point of communication.
 
 For example, if a grain is an aggregator of some counters or statistics that are reported by a large number of grains on a regular basis, one proven approach is to add a controlled number of intermediate aggregator grains and assign each of the reporting grains (using a modulo on a key or a hash) to an intermediate aggregator, so that the load is more or less evenly distributed across all intermediate aggregator grains that in their turn periodically report partial aggregates to the central aggregator grain.
 
-## How do I tear down a grain?
+### Can a single Orleans cluster run across multiple datacenters?
+
+Orleans clusters are currently limited to a single data center per cluster.
+Instead, since 1.3.0, you can consider a [multi-cluster deployment where clusters deployed to different datacenters form a single multi-cluster](Multi-Cluster/Overview.md). 
+
+### In what cases can a split brain (same grain activated in multiple silos at the same time) happen?
+
+During normal operations the Orleans runtime guarantees that each grain will have at most one instance in the cluster.
+The only time this guarantee can be violated is when a silo crashes or gets killed without a proper shutdown.
+In that case, there is a ~30 second (based on configuration) window where a grain can potentially get temporarily instantiated in more than one silo.
+
+You can find out more about how Orleans manages the clusters at [Cluster Management](Runtime-Implementation-Details/Cluster-Management.md) page.
+
+Also you can take a look at Orleans's [paper](http://research.microsoft.com/pubs/210931/Orleans-MSR-TR-2014-41.pdf) for a more detailed information, however you don't need to understand it fully to be able to write your application code.
+You just need to consider the rare possibility of having two instances of an actor while writing your application.
+
```

Comment:
Should we add that storage providers guarantee that no writes are overwritten without an activation first reading the most up-to-date state?

Reply:
I phrase it differently. See if you like it.